packages/manifold/test/gacha/ERC1155GachaLazyClaim.t.sol

```diff
@@ -0,0 +1,583 @@
+    creatorCore1.mintBaseNew(receivers, amounts, uris);
+    vm.stopPrank();
+
+    // mintreserving should have no effect
+    vm.startPrank(other);
+    example.mintReserve{ value: totalMintPrice * 2 }(address(creatorCore1), 1, 2);
+    vm.stopPrank();
+    vm.startPrank(other2);
+    example.mintReserve{ value: totalMintPrice }(address(creatorCore1), 1, 1);
+    vm.stopPrank();
+
+    vm.startPrank(creator);
+    claimP.tokenVariations = 3;
+    claimP.location = "arweaveHash2";
+    // create another gacha claim
+    example.initializeClaim(address(creatorCore1), 2, claimP);
+    vm.stopPrank();
+
+    assertEq("https://arweave.net/arweaveHash1/1", creatorCore1.uri(1));
+    assertEq("https://arweave.net/arweaveHash1/2", creatorCore1.uri(2));
+    assertEq("https://arweave.net/arweaveHash1/3", creatorCore1.uri(3));
+    assertEq("https://arweave.net/arweaveHash1/4", creatorCore1.uri(4));
+    assertEq("https://arweave.net/arweaveHash1/5", creatorCore1.uri(5));
+    assertTrue(
+      keccak256(bytes("https://arweave.net/arweaveHash1/6")) != keccak256(bytes(creatorCore1.uri(6))),
+      "URI should not match the specified value."
+    );
+    assertEq("https://arweave.net/arweaveHash2/1", creatorCore1.uri(7));
+    assertEq("https://arweave.net/arweaveHash2/2", creatorCore1.uri(8));
+    assertEq("https://arweave.net/arweaveHash2/3", creatorCore1.uri(9));
+  }
+
+  function testUpdateTokenURI() public {
+    vm.startPrank(creator);
+    uint48 nowC = uint48(block.timestamp);
+    uint48 later = nowC + 1000;
+    uint totalMintPrice = 1 + MINT_FEE;
+
+    IERC1155GachaLazyClaim.ClaimParameters memory claimP = IERC1155GachaLazyClaim.ClaimParameters({
+      storageProtocol: IGachaLazyClaim.StorageProtocol.ARWEAVE,
+      totalMax: 100,
+      startDate: nowC,
+      endDate: later,
+      tokenVariations: 5,
+      location: "arweaveHash1",
+      paymentReceiver: payable(creator),
+      cost: 1,
+      erc20: zeroAddress
+    });
+    example.initializeClaim(address(creatorCore1), 1, claimP);
+    
+    // tokens with original location
+    assertEq("https://arweave.net/arweaveHash1/1", creatorCore1.uri(1));
+    assertEq("https://arweave.net/arweaveHash1/2", creatorCore1.uri(2));
+    assertEq("https://arweave.net/arweaveHash1/3", creatorCore1.uri(3));
+    assertEq("https://arweave.net/arweaveHash1/4", creatorCore1.uri(4));
+    assertEq("https://arweave.net/arweaveHash1/5", creatorCore1.uri(5));
+
+    // update tokenURI
+    example.updateTokenURIParams(address(creatorCore1), 1, IGachaLazyClaim.StorageProtocol.ARWEAVE, "arweaveHashNEW");
+    assertEq("https://arweave.net/arweaveHashNEW/1", creatorCore1.uri(1));
+    assertEq("https://arweave.net/arweaveHashNEW/2", creatorCore1.uri(2));
+    assertEq("https://arweave.net/arweaveHashNEW/3", creatorCore1.uri(3));
+    assertEq("https://arweave.net/arweaveHashNEW/4", creatorCore1.uri(4));
+    assertEq("https://arweave.net/arweaveHashNEW/5", creatorCore1.uri(5));
+    vm.stopPrank();
+  }
+}
```

Comment:
assumes if we're updating the tokenURI we would update for ALL of the tokens
and if the creator updates the image even for one they would need to reupload the whole series